concent_api/core/payments/bankster.py

```diff
@@ -238,46 +201,47 @@ def finalize_payment(deposit_claim: DepositClaim) -> Optional[str]:
         elif available_funds_without_claims < deposit_claim.amount:
             deposit_claim.amount = available_funds_without_claims
 
-        # If the DepositClaim still exists at this point, Bankster uses SCI to create an Ethereum transaction.
-        if deposit_claim.concent_use_case == ConcentUseCase.FORCED_ACCEPTANCE:
-            ethereum_transaction_hash = service.force_subtask_payment(  # pylint: disable=no-value-for-parameter
-                requestor_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
-                provider_eth_address=deposit_claim.payee_ethereum_address,
-                value=deposit_claim.amount,
-                subtask_id=deposit_claim.subtask_id,
-            )
-        elif deposit_claim.concent_use_case == ConcentUseCase.ADDITIONAL_VERIFICATION:
-            subtask = Subtask.objects.filter(subtask_id=deposit_claim.subtask_id).first()  # pylint: disable=no-member
-            if subtask is not None:
-                task_to_compute = deserialize_message(subtask.task_to_compute.data.tobytes())
-                if task_to_compute.requestor_ethereum_address == deposit_claim.payer_deposit_account.ethereum_address:
-                    ethereum_transaction_hash = service.force_subtask_payment(  # pylint: disable=no-value-for-parameter
-                        requestor_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
-                        provider_eth_address=deposit_claim.payee_ethereum_address,
-                        value=deposit_claim.amount,
-                        subtask_id=deposit_claim.subtask_id,
-                    )
-                elif task_to_compute.provider_ethereum_address == deposit_claim.payer_deposit_account.ethereum_address:
-                    ethereum_transaction_hash = service.cover_additional_verification_cost(  # pylint: disable=no-value-for-parameter
-                        provider_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
-                        value=deposit_claim.amount,
-                        subtask_id=deposit_claim.subtask_id,
-                    )
-                else:
-                    assert False
-        else:
-            assert False
+    # If the DepositClaim still exists at this point, Bankster uses SCI to create an Ethereum transaction.
+    if deposit_claim.concent_use_case == ConcentUseCase.FORCED_ACCEPTANCE:
+        ethereum_transaction_hash = service.force_subtask_payment(  # pylint: disable=no-value-for-parameter
+            requestor_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
+            provider_eth_address=deposit_claim.payee_ethereum_address,
```

Comment:
quite frankly, it seems that whole `finalize_payment()` is called within some outer `atomic` block, so I assume all `with transaction.atomic(using='control')` should be removed from this function...

Reply:
quite frankly, it seems that whole `finalize_payment()` is called within some outer `atomic` block, so I assume all `with transaction.atomic(using='control')` should be removed from this function...

Reply:
I added an entry in the docstring. However I think it is a futile effort to add comment near each call, because locks (`select_for_update`) statements are pretty far from it (there is a few call levels between them).

Reply:
>  quite frankly, it seems that whole `finalize_payment()` is called within some outer `atomic` block, so I assume all `with transaction.atomic(using='control')` should be removed from this function...

Just as discussed on the hangout - I think those calls are done in `on_commit` hook which prevents transaction nesting in that case. But please check to make sure.

> However I think it is a futile effort to add comment near each call, because locks (`select_for_update`) statements are pretty far from it (there is a few call levels between them).

Maybe. Since these comments are extra information that his not otherwise available in the code (they're not just restating what we're doing) I think adding them would not hurt. Especially since we don't call this function all that many times. I just want someone who sees it to be aware that this function has such a big caveat.